src/main/java/com/elastic/support/diagnostics/DiagnosticService.java

```diff
@@ -58,9 +57,13 @@ public void exec(DiagnosticInputs diagnosticInputs, DiagConfig diagConfig, Map<S
 
             dc.runDiagnostic(ctx, chain);
 
+            if (ctx.isDocker()){
+                logger.warn("Identified Docker installations - bypassing log collection and system calls.");
```

Comment:
This is really saying that log collection and system calls were _already_ bypassed, right? If so, perhaps we should make this more of a past tense comment or put it wherever that is detected?

Reply:
A few suggestions weren't necessarily warranted - for instance, the REST call is displayed immediately prior to it's submittal so putting it in the error message would be redundant. However I did go back to look this over and decided that I wasn't all that happy with it. So I reworked the error handling in general to separate out unanticipated exceptions from one I explicitly throw. And use the message in those DiagnosticExceptions as the logged message displayed. Also, keep in mind there are two types of logging here. The normal info, warn, error variety which the user sees and the custom level, which contains stack traces and is only written to the internal log contained in the diagnostic.